docs/developer-guide/contributors-quickstart.md

```diff
@@ -28,9 +28,9 @@ cd src/github.com/argoproj &&
 git clone https://github.com/argoproj/argo-cd.git
 ```
 
-### Install Docker
+### Install Rancher Desktop
```

Comment:
I'm not so sure that we should remove Docker from the documentation, IMO it's fine to keep and add the caveats that you mentioned (not fully open-source etc), and then add Rancher as an alternative option.